designs/Prerelease-Option.md

```diff
@@ -0,0 +1,50 @@
+# Spec Name
+
+* Status: In Review
+* Author(s): [Martin Ruiz](https://github.com/martinrrm)
+* Issue: [4699](https://github.com/NuGet/Home/issues/4699) dotnet add package - support for NuGet "Pre" version
+
+## Problem Background
+
+Customers want to add the latest version of a package to their project, even if the latest version is prerelease.
+
+## Who are the customers
+
+All dotnet.exe customers
+
+## Goals
+
+Allow dotnet CLI users to install the latest version of a NuGet package, including prerelease versions. This can be accomplished by adding an option to the existing command `add package` that enables installing the last version, including prerelease versions.
+
+## Non-Goals
+
+## Solution
+
+Add the option `--prerelease` in the command `add package`. This command will get the latest version of the package available even if it is not a stable version.
+
+When searching for the latest version available, only listed packages will be considered.
+
+### Scenarios
+
+
+| Command | Available Versions | Output Version | Description |
+|---------|--------------------|------------------|-------------|
+| `dotnet.exe add Contoso.Library` | Contoso.Library 2.0.0 <br> Contoso.Library 3.0.0 <br> Contoso.Library 3.3.1-preview.3 <br> | 3.0.0 | latest stable version of package |
+| `dotnet.exe add Contoso.Library --prerelease` | Contoso.Library 2.0.0 <br> Contoso.Library 3.0.0 <br> Contoso.Library 3.3.1-preview.3 <br> | 3.3.1-preview.3 | latest version of package |
+| `dotnet.exe add Contoso.Library --prerelease --version 3.0.0` | Contoso.Library 2.0.0 <br> Contoso.Library 3.0.0 <br> Contoso.Library 3.3.1-preview.3 <br> | error | The user cannot use the `--prerelease` and `--version` commands at the same time |
```

Comment:
Should we error for `dotnet.exe add Contoso.Library --prerelease --version 3.0.0` or just ignore the --prerelease and let it work since the intent to install version 3.0.0 seems clear? I think that exact scenario is less likely, but I could potentially see someone doing  `dotnet.exe add Contoso.Library --prerelease --version 3.0.0-preview` thinking they need --prerelease to install it.

Erroring makes sense if that's the standard behavior of dotnet.exe in these kinds of scenarios.

Reply:
I think it's ok to error when the options provided are confusing and do not make sense. 

The https://docs.microsoft.com/en-us/nuget/reference/cli-reference/cli-ref-trusted-signers command throws a bunch of, hey you can't combine these switches errors.